### PR TITLE
Limit default Gaussian and parameter shift CV to single measurement with the new return type

### PR DESCRIPTION
**Pull request by rmoyard**
**Description of the Change:**
- Default Gaussian works with the new return type one but only for single measurement
- The parameter shift CV and other derivatives work with default Gaussian but only for single measurement

### CODE REVIEW

**Review by albi3ro — COMMENTED** [review]
Changelog and entry into deprecations page please.

**Comment by rmoyard** [comment]
@albi3ro Changelog added but there is no deprecation.

File: tests/returntypes/gradients/test_cv_gradients_new.py [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+            qml.Displacement(-0.5, 0.1, wires=0)
+            qml.Squeezing(0.5, -1.5, wires=0)
+            qml.Rotation(-1.1, wires=0)
+            return qml.expval(O(wires=0))
+
+        q = qml.QNode(circuit, gaussian_dev)
+
+        grad_F = qml.gradients.finite_diff(q)(par)
+        grad_A2 = qml.gradients.param_shift_cv(q, dev=gaussian_dev, force_order2=True)(par)
+        print(grad_F, grad_A2)
```
Comment (timmysilv) [inline tests/returntypes/gradients/test_cv_gradients_new.py:188]:
```suggestion
```

**Review by timmysilv — APPROVED** [review]
looks good!

File: pennylane/devices/default_gaussian.py [hunk 2]
```diff
@@ -888,3 +888,13 @@ def operations(self):
     @property
     def observables(self):
         return set(self._observable_map.keys())
+
+    def batch_execute(self, circuits):
+        if not qml.active_return():
+            return super().batch_execute(circuits)
+        for circuit in circuits:
+            if len(circuit.measurements) > 1:
+                raise qml.QuantumFunctionError("Default gaussian only support single measurements.")
```
Comment (albi3ro) [inline pennylane/devices/default_gaussian.py:897]: What happens if such a circuit passed to execute instead of batch_execute?

Comment (rmoyard) [reply pennylane/devices/default_gaussian.py:897]: Do you want me to add the error in execute as well?

Comment (rmoyard) [reply pennylane/devices/default_gaussian.py]: I changed the execute to have it in. Codefactor is omplaining but it is not a problem as pylint is passing.

**Review by albi3ro — APPROVED** [review]

**Pull request merged**